charlib/liberty/liberty.py

```
@@ -93,16 +123,21 @@ def subgroups_with_name(self, name: str):
+        :param identifier: (Optional) The subgroup identifier to search for. Ignored if not
+                           specified.
+        :param **attributes: (Optional) Name-value pairs corresponding to subgroup attributes
+                             to match.
         """
-        for group in self.groups.values():
-            if group.identifier == identifier:
-                yield group
-            elif group.groups:
-                yield from group.subgroups_with_identifier(identifier)
+        for group in self.subgroups_with_name(name):
+            if identifier and not group.identifier == identifier:
```

Ruff (0.15.20)

[warning] Use `group.identifier != identifier` instead of `not group.identifier == identifier`

Replace with `!=` operator

(SIM201)